sntools/interaction_channels/o16nc_n.py

```diff
@@ -0,0 +1,218 @@
+"""Implementation of:
+    nu + 16O -> nu' + 16O*, 16O* -> 15O + n.
+
+Based on data provided in Suzuki et al. 2018 (Phys. Rev. C 98, 034613).
+A spline fit is used to obtain cross-section as a function of neutrino energy.
+Some data points have been estimated from a by-eye fit in order to create a 
+smooth spline fit. The threshold energy of the interaction was also estimated 
+from a by-eye fit as no explicit threshold energy could be found.
+
+To determine the the differential cross section dSigma/dE (eNu, eE) from the
+total cross section, we approximate a DiracDelta function with one that is
+2*epsilon wide and 1/(2*epsilon) high, so that the integral is 1.
+"""
+
+from sntools.event import Event
+from sntools.interaction_channels import BaseChannel
+from scipy.interpolate import interp1d
+#import numpy as np
+import random
+
+""" This table shows the partial cross-section of 16O(nu, nu' X). It is taken 
+directly from Suzuki et al. 2018 with an estimated threshold energy data point to
+ensure a smooth spline fit. (cross-section units 10^-42 cm^2)
+(eNu, gamma, n, p, d, pp, 3H, 3He, alpha, alpha_n, alpha_p, total)
+"""
+
+e_thr = 19.5  # approximate energy threshold of neutron emission with no gamma emission (MeV)
+mN = 939.7 # neutron mass (MeV)
+'''e_thr_g = e_thr + 6.18  # energy threshold of neutron emission with gamma emission (MeV)'''
+epsilon = 0.001  # for approximating DiracDelta distribution below
+
+# List of neutrino flavors ("e", "eb", "x", "xb") that interact in this channel.
+possible_flavors = ("e", "eb", "x", "xb")
+
+x_sec_data = [(e_thr, None    , 0.0    , None    , None    , None    , None    , None    ,
```

Comment:
It looks like you’re going to a lot of effort here and over the next ~60 lines of code to make this extremely general and capable of supporting all emission channels—but only one of them is actually used in the code, unless I’m missing something?
I think I would prefer to remove the code for those other channels here, to make it a lot shorter and easier to understand.